Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Auth/Login/LogInViewController.swift

```diff
@@ -0,0 +1,163 @@
+        signupButton.titleLabel?.font = UIFont.systemFont(ofSize: 22, weight: .bold)
+        
+        Utilities.styleTextField(textfield: emailTextField)
+        Utilities.styleTextField(textfield: passwordTextField)
+        Utilities.styleLogInButton(button: loginButton)
+        Utilities.styleShadowButton(button: loginButton)
+        
+        emailTextField.attributedPlaceholder = NSAttributedString(string: "", attributes: [
+            .foregroundColor: UIColor.black,
+            .font: UIFont.boldSystemFont(ofSize: 16.0)
+        ])
+        passwordTextField.attributedPlaceholder = NSAttributedString(string: "", attributes: [
+            .foregroundColor: UIColor.black,
+            .font: UIFont.boldSystemFont(ofSize: 16.0)
+        ])
+    }
+    
+    @IBAction private func loginTapped(_ sender: Any) {
+        let userEmail = emailTextField.text
+        let userPassword = passwordTextField.text
+        
+        if userEmail == "" || userPassword == "" {
+            alertMessage(userMessage: "All fields are required")
+        } else {
+            let mail = isValidEmail(emailID: userEmail)
+            if mail == false {
+                alertMessage(userMessage: "Please enter a valid email addres!")
+                emailLabel.textColor = .red
+            } else {
+                if isValidPassword(password: userPassword) == false {
+                    alertMessage(userMessage: "Please enter a valid password!")
+                    emailLabel.textColor = .colorText
+                    passwordLabel.textColor = .red
+                } else {
+                    alertMessage(userMessage: "Success!")
+                    passwordLabel.textColor = .colorText
+                }
+            }
+        }
+    }
+    
+    @IBAction private func signupTapped(_ sender: Any) {
+    }
+
+    //Email validation
+    private func isValidEmail(emailID: String?) ->Bool {
+        let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+        let emailTest = NSPredicate(format: "SELF MATCHES %@", emailRegEx)
+        return emailTest.evaluate(with: emailID)
+    }
+    
+    //Password validation
+    private func isValidPassword(password: String?) ->Bool{
+        let passwordReg = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)[a-zA-Z\\d]{8,}$"
+        let passwordTesting = NSPredicate(format: "SELF MATCHES %@", passwordReg)
+        return passwordTesting.evaluate(with: password)
+    }
+    
+    //Alert
+    private func alertMessage(userMessage: String) {
+        let myAllert = UIAlertController(title: "Something went wrong", message: userMessage, preferredStyle: UIAlertController.Style.alert)
+        let ok = UIAlertAction(title: "OK", style: UIAlertAction.Style.default, handler:  nil)
+        myAllert.addAction(ok)
+        present(myAllert, animated: true, completion: nil)
+    }
+   
+    class Utilities {
+    //Style the text fields
+    final class func styleTextField(textfield:UITextField) {
```

Comment:
Remove unnecessary comments (style the text fields, that can be understood from the method's name)